### DISCUSSION TITLE
Timeout issue when creating moose environment.

### DISCUSSION THREAD

**Discussion by BrownieLL** [idaholab/moose#25148] [Q&A Getting Started]
Hello everyone,

I'm reaching out as a newcomer to both Moose and the Linux system. I'm currently working on setting up Moose on the Windows Subsystem for Linux (WSL) following the official guidelines. However, I'm encountering issues while trying to build the Moose environment using the "mamba create -n moose moose-dev" command. The process fails multiple times with:

_Looking for: ['moose-dev']

https://conda.software.inl.gov/public/linux-64      ??.?MB @  ??.?MB/s 0 failed 10.0s
Download error (28) Timeout was reached [https://conda.software.inl.gov/public/noarch/repodata.json]
Could not resolve host: conda.software.inl.gov_

Interestingly, there are instances when I manage to successfully download some of the packages, but the process fails due to timeouts partway through. I'm a bit baffled by this situation since I'm in Australia and I believe there shouldn't be any restrictions on website access from my location.

I'm hoping that someone with more experience could kindly assist me in resolving this issue. Your insights and guidance would be greatly appreciated.

Thank you,
Zinn

**Comment by GiudGiud** [comment]
Hello

We've had network issues earlier this week. I did not hear that they hit our conda packages but maybe they did.
Do you have another machine you could try from? Just to check if the issues are tied to your machine WSL or not

Guillaume

**Reply by BrownieLL** [reply]
Hi, Guillaume.

Thanks for the information.

Yes, I have try another machine, and it also meet the same issue. After numerous attempts, I was finally able to successfully install all those packages.

Thank you!

Best,
Zinn

**Reply by milljm** [reply]
There was another user who had the same issue. They were able to solve the timeouts by increasing the threshold:

```bash
conda config --set remote_read_timeout_secs new_time_out
```
where `new_time_out` was something greater than the default `60` seconds. Something like this definitely belongs on our [troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) page...

**Reply by milljm** [reply]
Created an issue to add this little hint to our troubleshooting page: https://github.com/idaholab/moose/issues/25152

**Reply by BrownieLL** [reply]
I appreciate your suggestions, but the solution provided didn't yield the desired results in my case. Despite executing the command "conda config --set remote_read_timeout_secs 120," the timeout limitation persists at 10 seconds. Could this issue potentially be attributed to my use of Mamba for package installation instead of Conda?

**Reply by milljm** [reply]
It could also be related to another issue being reported by this individual: https://github.com/idaholab/moose/discussions/25149

Using a web browser, are you able to peruse to this meta URL: https://conda.software.inl.gov/public/noarch/repodata.json
If so, what happens when you use `curl` instead of a web browser (testing that your network connection works from a terminal session):

```bash
curl https://conda.software.inl.gov/public/noarch/repodata.json
```

**Reply by BrownieLL** [reply]
No, I can't connect to that URL now, but a few hours ago, I successfully connected. I am in Australia.

**Reply by BrownieLL** [reply]
I'm currently unable to access the official website of MOOSE as well (https://mooseframework.inl.gov/).

**Reply by milljm** [reply]
Thank you for reporting! Looks like our servers are somehow blocking certain ranges of IPs... I've reported the issue to our network department (and we will voice the importance of getting this solved!).

**Reply by milljm** [reply]
@BrownieLL, can you post a screen shot of what occurs when you attempt to navigate to: https://conda.software.inl.gov/public/noarch/repodata.json

This is a request from my IM department. We would like some quantifiable artifacts to proceed with

**Reply by BrownieLL** [reply]
Sorry, I was sleeping at that time. I can access to that URL now.

**Reply by GiudGiud** [reply]
Nice
